benchmark-utils_test.go

```diff
@@ -139,21 +129,57 @@ func runGetObjectBenchmark(b *testing.B, obj ObjectLayer, objSize int) {
 
 }
 
-// randomly picks a character and returns its equivalent byte array.
-func getRandomByte() []byte {
-	const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
-	// seeding the random number generator.
-	rand.Seed(time.Now().UnixNano())
-	var b byte
-	// pick a character randomly.
-	b = letterBytes[rand.Intn(len(letterBytes))]
-	return []byte{b}
-}
+// Returns os.File and MD5 hash string of required size of random bytes.
+func mustGetRandomReader(size int) (file *os.File, hash string) {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	hasher := md5.New()
+
+	// Create temporary file.
+	tmpfile, err := ioutil.TempFile("", "minio-bench")
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(tmpfile.Name())
+
+	var buf []byte
+	if size < readSizeV1 {
+		buf = make([]byte, size)
+	} else {
+		buf = make([]byte, readSizeV1)
+	}
+
+	// Write to the temporary file.
+	bytesLeft := size
+	n := 0
+	for bytesLeft > 0 {
+		if bytesLeft > readSizeV1 {
+			n = readSizeV1
+		} else {
+			n = bytesLeft
+		}
+
+		r.Read(buf[:n])
```

Comment:
> The problem will be when writing 1G worth of rand.Reader, will take really long long time. bytes.Repeat is completely fine if we can see how to avoid them being reported as part of the allocs. 

Agree. The b.ReportAllocs() are done after the setup, I had assumed that this shouldn't take previous allocations into consideration.